lib/ansible/modules/storage/netapp/na_cdot_snapshot.py

```diff
@@ -0,0 +1,238 @@
+from ansible.module_utils._text import to_native
+import ansible.module_utils.netapp as netapp_utils
+
+
+HAS_NETAPP_LIB = netapp_utils.has_netapp_lib()
+
+
+class NetAppCDOTSnapshot(object):
+
+    def __init__(self):
+
+        self._size_unit_map = dict(
+            bytes=1,
+            b=1,
+            kb=1024,
+            mb=1024 ** 2,
+            gb=1024 ** 3,
+            tb=1024 ** 4,
+            pb=1024 ** 5,
+            eb=1024 ** 6,
+            zb=1024 ** 7,
+            yb=1024 ** 8
+        )
+
+        self.argument_spec = netapp_utils.ontap_sf_host_argument_spec()
+        self.argument_spec.update(dict(
+            state=dict(required=True, choices=['present', 'absent']),
+            name=dict(required=True, type='str'),
+            volume_name=dict(required=True, type='str'),
+            vserver=dict(required=True, type='str', default=None),
+        ))
+
+        self.module = AnsibleModule(
+            argument_spec=self.argument_spec,
+            required_if=[
+                ('state', 'present', ['volume_name'])
+            ],
+            supports_check_mode=True
+        )
+
+        p = self.module.params
+
+        # set up state variables
+        self.state = p['state']
+        self.name = p['name']
+        self.vserver = p['vserver']
+        self.volume_name = p['volume_name']
+
+        if HAS_NETAPP_LIB is False:
+            self.module.fail_json(msg="the python NetApp-Lib module is required")
+        else:
+            self.server = netapp_utils.setup_ontap_zapi(module=self.module, vserver=self.vserver)
+
+    def get_snapshot(self):
+        """
+        Return details about the snapshot
+        :param:
+            name : Name of the smapshot
+            volume_name: Name of the containing volume
+
+        :return: Details about the snapshot. None if not found.
+        :rtype: dict
+        """
+        snapshot_list = netapp_utils.zapi.NaElement('snapshot-get-iter')
+        snapshot_info = netapp_utils.zapi.NaElement('snapshot-info')
+        snapshot_info.add_new_child('name', self.name)
+        snapshot_info.add_new_child('volume', self.volume_name)
+
+        query = netapp_utils.zapi.NaElement('query')
+        query.add_child_elem(snapshot_info)
+
+        snapshot_list.add_child_elem(query)
+
+        result = self.server.invoke_successfully(snapshot_list, True)
+
+        return_value = None
+
+        if result.get_child_by_name('num-records') and \
+                int(result.get_child_content('num-records')) >= 1:
+
+            snapshot_info = result.get_child_by_name(
+                'attributes-list').get_child_by_name(
+                    'snapshot-info')
+            # check if snapshot name & volume match
+            returned_name = snapshot_info.get_child_content('name')
+            returned_volume = snapshot_info.get_child_content('volume')
+
+            if returned_name == self.name and returned_volume == self.volume_name:
```

Comment:
Why is this check necessary, since you're querying by self.name and self.volume_name, you'll only get the snapshots?

Reply:
You're right, as I'm directly querying the volume, I already know that this matches when I get a result back.